paddle/fluid/operators/mkldnn/activation_mkldnn_op.cc

```diff
@@ -99,86 +99,56 @@ void eltwise_forward(const framework::ExecutionContext &ctx,
   auto src_format =
       src_tz.size() == 2 ? mkldnn::memory::format::nc : x->format();
 
-  const std::string key = gethash(src_tz, algorithm);
-  const std::string key_src_data =
-      key + ctx.op().Output("Out") + "@eltwise_fwd_src_data";
-  const std::string key_src_layout =
-      key + ctx.op().Output("Out") + "@eltwise_fwd_src_layout";
-  const std::string key_with_layout = key + std::to_string(src_format);
-  const std::string key_src_mem = key_with_layout + "@eltwise_fwd_src_mem";
-  const std::string key_dst_mem = key_with_layout + "@eltwise_fwd_dst_mem";
-  const std::string key_fwd = key_with_layout + "@eltwise_fwd";
-  const std::string key_fwd_pd = key_with_layout + "@eltwise_fwd_pd";
-
   bool is_test = ctx.Attr<bool>("is_test");
 
+  // TODO(jczaja): When adding leaky-relu , swish , elu make sure to extend key
+  // with alpha, beta
+  std::string key = platform::MKLDNNHandler::GetHash(
+      src_tz, std::to_string(algorithm) + ctx.op().Output("Out"));
+
+  // TODO(jczaja): Make it Thread safe
   // save input data and layout to be referred in backward path
+  const std::string key_src_data = key + "@eltwise_fwd_src_data";
+  const std::string key_src_layout = key + "@eltwise_fwd_src_layout";
+  // Just in case some int8 models are run interchangebly
+  // with float models then format maybe diffrent
+  key += std::to_string(src_format);
+  const std::string key_src_mem = key + "@eltwise_fwd_src_mem";
   auto p_src_data = std::make_shared<const T *>(x_data);
```

Comment:
why use explicit setblob here instead of an api in ActivationMKLDNNHandler, e.g. AcquireSrcLayout?

Reply:
I left this intentionally like this as I need to resolve #17960 to have this done properly so those lines will be changes in next PRs.

Reply:
understood your idea. so you want to keep this key without tid, right?
I remember there is another INT8 conv performance drop issue which may be related with key without tid, by my investigation at that time, it is due to not inserting tid into key, it was fixed with tid into key, so be careful on performance if use this method.

Reply:
Currently this code is refactoring. Once #17960 is cleared I will modify changes you pointed.